### PR TITLE
Migrate away from poetry

### PR DESCRIPTION
**Pull request by FloLey**
move away from poetry as it was causing issues.

### CODE REVIEW

**Comment by FloLey** [comment]
@sanderr Do I need to do something more than this? I didn't do it as in the PR of Darius as I needed something for the extras hence the setup.py.

File: .ci/Jenkinsfile-integration-test [hunk 1]
```diff
@@ -31,28 +30,18 @@ pipeline {
                             rm -rf ${WORKSPACE}/env
                             python3 -m venv ${WORKSPACE}/env
                             ${WORKSPACE}/env/bin/pip install -U pip
-                            ${WORKSPACE}/env/bin/pip install --pre irt
-                            python_version=$(${WORKSPACE}/env/bin/irt query-toml-file --file ./pyproject.toml --query "tool.poetry.dependencies.python"| sed "s/\\^//g")
```
Comment (sanderr) [inline .ci/Jenkinsfile-integration-test:35]: I think we still want to keep this (modified of course)

File: pyproject.toml [hunk 2]
```diff
@@ -1,65 +1,72 @@
...
+    "pytest-postgresql==5.0.0",
+    "psycopg==3.1.9",
+    "tornado>=6.1,<7.0"
+]
+async = [
+    "pytest-asyncio==0.21.0",
+    "pytest-timeout==2.1.0"
+]
+pytest = [
+    "pytest-env==0.8.2",
```
Comment (sanderr) [inline pyproject.toml:54]: I hope dependabot can manage with some dependencies occurring more than once.

**Review by sanderr — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 2 above)
Comment (FloLey) [reply pyproject.toml:54]: I hope so to as well

**Review by sanderr — APPROVED** [review]

File: .ci/Jenkinsfile-integration-test [hunk 3]
```diff
@@ -32,27 +31,26 @@ pipeline {
                             python3 -m venv ${WORKSPACE}/env
                             ${WORKSPACE}/env/bin/pip install -U pip
                             ${WORKSPACE}/env/bin/pip install --pre irt
-                            python_version=$(${WORKSPACE}/env/bin/irt query-toml-file --file ./pyproject.toml --query "tool.poetry.dependencies.python"| sed "s/\\^//g")
+                            python_version=$(${WORKSPACE}/env/bin/irt query-toml-file --file ./pyproject.toml --query "project.requires-python"| sed "s/>=//g")
                             python_binary="python${python_version}"
-
+                            echo ${python_binary}
```
Comment (sanderr) [inline .ci/Jenkinsfile-integration-test:36]:
```suggestion
```

File: pyproject.toml [hunk 4]
```diff
@@ -1,65 +1,72 @@
...
+    "pytest-xdist==3.3.1",
+    "pytest-sugar==0.9.7",
+    "pytest-instafail==0.5.0"
+]
 sphinx = [
-    "inmanta-sphinx", "sphinx-argparse", "sphinx-autodoc-annotation", "sphinx-tabs", "sphinx", "sphinxcontrib-serializinghtml",
-    "sphinxcontrib-redoc", "sphinx-click", "recommonmark", "myst_parser"
+    "inmanta-sphinx==1.7.0",
+    "sphinx-argparse==0.4.0",
+    "sphinx-autodoc-annotation==1.0-1",
```
Comment (Hugo-Inmanta) [inline pyproject.toml:64]:
```suggestion
    "sphinx-autodoc-annotation==1.0.1",
```

File: pyproject.toml [hunk 5]
```diff
@@ -1,65 +1,72 @@
...
+    "sphinx-tabs",
+    "sphinx==6.2.1",
+    "sphinxcontrib-serializinghtml==1.1.5",
+    "sphinxcontrib-redoc==1.6.0",
+    "sphinx-click==4.4.0",
+    "recommonmark",
+    "myst_parser==2.0.0"
```
Comment (Hugo-Inmanta) [inline pyproject.toml:71]: Where do these version constraints numbers come from ? Why don't we also pin the others ? i.e. sphinx-tabs, recommonmark etc ?

**Review by Hugo-Inmanta — CHANGES_REQUESTED** [review]
Isn't this going to break for iso4 which is python 3.6 only ?

File: pyproject.toml (hunk 4 above)
Comment (FloLey) [reply pyproject.toml:64]: This is how the requirement was specified before. I don't think we want to change this. (see line 40 before the change)

File: pyproject.toml (hunk 5 above)
Comment (FloLey) [reply pyproject.toml:71]: They come form here: https://github.com/inmanta/inmanta-dev-dependencies/pull/306/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711L26

some are pinned, some are not.

**Comment by FloLey** [comment]
> Isn't this going to break for iso4 which is python 3.6 only ?

@Hugo-Inmanta inmanta-dev-dependencies is only run against master: https://jenkins.inmanta.com/job/development-tools/job/inmanta-dev-dependencies/

File: pyproject.toml (hunk 4 above)
Comment (Hugo-Inmanta) [reply pyproject.toml:64]: Ah indeed: https://pypi.org/project/sphinx-autodoc-annotation/1.0-1/#history

File: pyproject.toml (hunk 5 above)
Comment (Hugo-Inmanta) [reply pyproject.toml:71]: I see thanks

File: pyproject.toml (hunk 4 above)
Comment (FloLey) [reply pyproject.toml:64]: Ho I thought it was just a way to say version 1.0 or 1.1. but it seems it is just a weird versioning than?

**Comment by Hugo-Inmanta** [comment]
> Isn't this going to break for iso4 which is python 3.6 only ?

Ok I realize now this question makes no sense

**Review by Hugo-Inmanta — APPROVED** [review]

**Pull request merged**